src/library_glemu.js

```diff
@@ -3415,9 +3489,27 @@ var LibraryGLEmulation = {
   glReadBuffer: function() { throw 'glReadBuffer: TODO' },
 #endif
 
-  glLightfv: function() { throw 'glLightfv: TODO' },
-  glLightModelfv: function() { throw 'glLightModelfv: TODO' },
-  glMaterialfv: function() { throw 'glMaterialfv: TODO' },
+  glClipPlane: function(pname, param) { // partial support, TODO
+    if ((pname >= 0x3000) && (pname < 0x3006)  /* GL_CLIP_PLANE0 to GL_CLIP_PLANE5 */) {
+      var clipPlaneId = pname - 0x3000;
+
+      GLEmulation.clipPlaneEquation[clipPlaneId][0] = {{{ makeGetValue('param', '0', 'double') }}};
+      GLEmulation.clipPlaneEquation[clipPlaneId][1] = {{{ makeGetValue('param', '8', 'double') }}};
+      GLEmulation.clipPlaneEquation[clipPlaneId][2] = {{{ makeGetValue('param', '16', 'double') }}};
+      GLEmulation.clipPlaneEquation[clipPlaneId][3] = {{{ makeGetValue('param', '24', 'double') }}};
+
+      // apply inverse transposed current modelview matrix when setting clip plane
+      var tmpMV = GLImmediate.matrixLib.mat4.create(GLImmediate.matrix[0]);
+      GLImmediate.matrixLib.mat4.inverse(tmpMV);
+      GLImmediate.matrixLib.mat4.transpose(tmpMV);
+      GLImmediate.matrixLib.mat4.multiplyVec4(tmpMV, GLEmulation.clipPlaneEquation[clipPlaneId]);
+    }
+  },
+
+  glLightfv: function() {}, //{ throw 'glLightfv: TODO' },
+  glLightModelf: function() {}, //{ throw 'glLightModelf: TODO' },
+  glLightModelfv: function() {}, //{ throw 'glLightModelfv: TODO' },
+  glMaterialfv: function() {}, //{ throw 'glMaterialfv: TODO' },
```

Comment:
it would be good to warn here at least, when in `ASSERTIONS`, that these are not fully implemented.

Reply:
Yes. That was not supposed to be in the initial PR. See comment below.